compiler/aarch64/codegen/FPTreeEvaluator.cpp

```diff
@@ -818,12 +818,60 @@ OMR::ARM64::TreeEvaluator::dcmpleEvaluator(TR::Node *node, TR::CodeGenerator *cg
    return fcmpHelper(node, TR::CC_LS, true, cg);
    }
 
+static TR::Register *
+floatThreeWayCompareHelper(TR::Node *node, bool isDouble, bool isCmpl, TR::CodeGenerator *cg)
+   {
+   // dcmpl/fcmpl: 1 if v1 > v2, 0 if v1 == v2, -1 if v1 < v2 or unordered
+   // dcmpg/fcmpg: 1 if v1 > v2 or unordered, 0 if v1 == v2, -1 if v1 < v2
+
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Node *secondChild = node->getSecondChild();
+   TR::Register *src1Reg = cg->evaluate(firstChild);
+   TR::Register *src2Reg = cg->evaluate(secondChild);
+   TR::Register *trgReg = cg->allocateRegister();
```

Comment:
Can you check the reference counts on the source registers and re-use one of them if their live range ends after the first compare?  Both src1 and src2 are only used before the first use of the target register.

Reply:
You cannot re-use src1Reg nor src2Reg in this case, because the source registers are FP registers and the target register is a GP register.

Reply:
Ah, right.  Of course.